Fixed #35829 -- Updated admin icons using Font Awesome Free version 6.7.2.

pablobn opened the pull request:
#### Trac ticket number

ticket-35829

#### Branch description
Update Font Awesome icons in django/contrib/admin/static/admin/img to the latest version (6.7.1) to address potential licensing issues. Revised the README and LICENSE files to reflect the updates and ensure compliance with Font Awesome's licensing terms. Pending clarification on the usage and replacement of icons in the gis folder. Related to ticket #35829.

stefan6419846 commented on django/contrib/admin/static/admin/img/README.txt:
```diff
@@ -1,7 +1,18 @@
-SVG icons source: https://github.com/encharm/Font-Awesome-SVG-PNG
-Font-Awesome-SVG-PNG is licensed under the MIT license (see file license
-in current folder).
+All icons in this directory are provided by [Font Awesome Free](https://fontawesome.com), version 6.x.
```
Why not providing a specific version?

pablobn replied:
Because specifying version 6.x provides flexibility to choose any icon from the sixth major version. I could also include 5.x if we want to support icons from that version as well.

ulgens replied:
+1 for being specific here. "6.x" creates the expectation of being always up to date with new 6.x releases, which I don't think will be the case.

pablobn replied:
"All icons are taken from Font Awesome (https://fontawesome.com/) project."
"All icons in this directory are provided by [Font Awesome Free](https://fontawesome.com)"

On the other hand, do you think these changes are reasonable?

stefan6419846 replied:
The issue is that as soon as we omit a specific version, license compliance can become hard. Django essentially bundles a modified copy of a third-party library and linking to the unmodified library actually used here inside an own SBOM can be desired. This is much easier if Django provides the exact upstream version.

pablobn replied:
However, this primarily concerns the icons, which are elements that are unlikely to change rapidly.

stefan6419846 commented on django/contrib/admin/static/admin/img/README.txt:
```diff
@@ -1,7 +1,18 @@
+- This license allows you to use, modify, and distribute the icons, provided proper attribution is given.
+
+### Attribution:
+To comply with the CC BY 4.0 license, please include the following attribution in your project:
```
```suggestion
To comply with the CC-BY-4.0 license, please include the following attribution in your project:
```

Additionally: Some text seems to be missing here.

stefan6419846 commented on django/contrib/admin/static/admin/img/README.txt:
```diff
@@ -1,7 +1,18 @@
+
+## Usage
+
+You may use, modify, and distribute the icons in this repository in compliance with the [Creative Commons Attribution 4.0 International (CC BY 4.0)](https://creativecommons.org/licenses/by/4.0/) license.
```
```suggestion
You may use, modify, and distribute the icons in this repository in compliance with the [Creative Commons Attribution 4.0 International (CC-BY-4.0)](https://creativecommons.org/licenses/by/4.0/) license.
```

stefan6419846 commented on django/contrib/admin/static/admin/img/README.txt:
```diff
@@ -1,7 +1,18 @@
+- This license allows you to use, modify, and distribute the icons, provided proper attribution is given.
+
+### Attribution:
+To comply with the CC BY 4.0 license, please include the following attribution in your project:
```
The actual text still seems to be missing?

pablobn replied:
Yes, I'm going to delete that part

knyghty commented:
With the caveat that I haven't (yet) looked at this in a browser, I notice many of the icons have a tighter crop than the old one. This is much better for various reasons, but I think this may reduce things like the padding of some widgets / between some icons / etc.

I will try to review this at some point.

knyghty commented:
For width/height, yes it should be addressed at least to avoid layout shift.

I don't use GIS often, there is a field that renders a map so it could be used in this, but not sure where or how offhand.

nessita commented:
@pablobn Hello, thank you so much for this work!

Were you able to check where/how the GIS icons are used?

pablobn commented:
> @pablobn Hello, thank you so much for this work!
> 
> Were you able to check where/how the GIS icons are used?

Hi @nessita, I'm sorry i didn't have too much time, i don't know where are those icons used.

pablobn commented:
Hello, @nessita. I was reviewing the Django project and noticed that the folder was introduced in version 1.0.x. Its usage seems to have been limited to a single .html file located at django/contrib/gis/template/gis/admin/openlayer.html. However, in version 5.0.x, the folder django/contrib/gis/template/gis/admin/ was removed, so I understand it is no longer necessary.

claudep commented:
The GIS icons are removed in PR #19091. Thanks for noticing!

nessita reviewed (commented):
Thank you @pablobn for this work and for your patience. :star2: 

I have reviewed this PR and I think it's ready to be landed, with the only caveat that perhaps, if you would have any availability, we could update to font awesome free 6.7.2?

pablobn commented:
Perfect @nessita. I will change it next week that I have free time

> Thank you @pablobn for this work and for your patience. 🌟
> 
> I have reviewed this PR and I think it's ready to be landed, with the only caveat that perhaps, if you would have any availability, we could update to font awesome free 6.7.2?

Perfect @nessita, I will update it next week

thibaudcolas commented on django/contrib/admin/static/admin/img/LICENSE:
Personally I think it’d be easier to understand the licensing of the icons by referring to the comments within the icons files. As far as I understand that’s the intention from FontAwesome? As-is I think it makes it harder to do your licensing due diligence, due to the file containing SIL OFL and MIT references which are irrelevant to what we’re doing here. And even the relevant Attribution information, I’d rather reference the link provided by FontAwesome, than this copy.

thibaudcolas commented on django/contrib/admin/static/admin/img/README.txt:
```diff
@@ -1,7 +1,15 @@
-All icons are taken from Font Awesome (https://fontawesome.com/) project.
-The Font Awesome font is licensed under the SIL OFL 1.1:
-- https://scripts.sil.org/OFL
+## Icons License
```
```suggestion
# Icons License
```

Not a must but if we use headings formatting it’d be nice to use correct levels here and below.

thibaudcolas commented on django/contrib/admin/static/admin/img/README.txt:
```diff
@@ -1,7 +1,15 @@
-SVG icons source: https://github.com/encharm/Font-Awesome-SVG-PNG
-Font-Awesome-SVG-PNG is licensed under the MIT license (see file license
-in current folder).
+All icons in this directory are provided by [Font Awesome Free](https://fontawesome.com), version 6.7.1
```
I don’t think that’s the case but could be wrong? The files in the `gis` subdirectory don’t look like Font Awesome to me. I could also see in the future us adopting other icons and placing them in this folder as it’s a generic `img` folder, not `fontawesome`.

thibaudcolas commented on django/contrib/admin/static/admin/img/calendar-icons.svg:
```diff
@@ -2,62 +2,40 @@
-    <g
-       id="previous">
+   xmlns="http://www.w3.org/2000/svg">
+   <!-- 
+   All icons are provided by Font Awesome Free, version 6.7.1 
+   Website: https://fontawesome.com 
+   License: Creative Commons Attribution 4.0 International (CC BY 4.0)
+   For more details, visit: https://fontawesome.com/license/free 
+   Copyright 2024 Fonticons, Inc.
+   -->
```
Personally I’d find it easier if we kept the original licensing comments from FontAwesome as-is.

pablobn commented:
In this new update, I’ve made the following changes:

- Removed the gis folder, as it was not being used.

- Deleted the LICENSE file, following @thibaudcolas’s suggestion, since license information for the icons can be found directly in the file comments or on Font Awesome’s official website.

- Added a contrib.md file to document the added resources. I’d appreciate any feedback on whether this file should be modified or expanded.

- Updated the icons to Font Awesome version 6.7.2.

- Adjusted the size of inline-deletelink.svg to make it more appropriate.

- Included the icon name, family, and style as comments within each SVG file to make it easier to identify and trace the source of each icon if needed in the future.

nessita commented:
> The GIS icons are removed in PR #19091. Thanks for noticing!

@claudep Could you help me understand a bit more when/where these icons were removed? Right now, what's left in the source tree is:
```
        django/contrib/gis/static/gis/img/draw_line_off.svg
        django/contrib/gis/static/gis/img/draw_line_on.svg
        django/contrib/gis/static/gis/img/draw_point_off.svg
        django/contrib/gis/static/gis/img/draw_point_on.svg
        django/contrib/gis/static/gis/img/draw_polygon_off.svg
        django/contrib/gis/static/gis/img/draw_polygon_on.svg
```

These files are still used by `django/contrib/gis/static/gis/css/ol3.css`:
```css
.type-Point {
    background-image: url("../img/draw_point_off.svg");
    right: 5px;
}
.type-Point.type-active {
    background-image: url("../img/draw_point_on.svg");
}

.type-LineString {
    background-image: url("../img/draw_line_off.svg");
    right: 30px;
}
.type-LineString.type-active {
    background-image: url("../img/draw_line_on.svg");
}

.type-Polygon {
    background-image: url("../img/draw_polygon_off.svg");
    right: 55px;
}
.type-Polygon.type-active {
    background-image: url("../img/draw_polygon_on.svg");
}
```

And the `ol3.css` is used in `django/contrib/gis/forms/widgets.py`, and so is the `django/contrib/gis/static/gis/js/OLMapWidget.js` javascript file that has code that seems to require the CSS classes:
```js
        const element = document.createElement('div');
        element.className = 'switch-type type-' + options.type + ' ol-control ol-unselectable';
        if (options.active) {
            element.classList.add("type-active");
        }
```
I'm not sure how to trigger the JS functions to fully see the icons used, so I would appreciate some further clarification on this. Thank you!

claudep commented:
Those icons should not be removed. They are used in the map widgets used to display geometric values. To show how/where they are used, you just need to display a GIS-enabled model in the admin.

nessita commented:
> Those icons should not be removed. They are used in the map widgets used to display geometric values. To show how/where they are used, you just need to display a GIS-enabled model in the admin.

Thank you! That makes more sense. From the commentary above it seemed as if we could remove all GIS icons.

nessita reviewed (approved):
Thank you @pablobn! I pushed some cleanups. I'll merge when CI is green.

ngnpope commented on django/contrib/admin/static/admin/img/README.md:
```diff
@@ -0,0 +1,78 @@
+# Information about icons in this directoy
```
A little typo:

```suggestion
# Information about icons in this directory
```
